src/compiler/checker.ts

```diff
@@ -26627,11 +26625,12 @@ namespace ts {
 
         function checkSpreadPropOverrides(type: Type, props: SymbolTable, spread: SpreadAssignment | JsxSpreadAttribute) {
             for (const right of getPropertiesOfType(type)) {
-                const left = props.get(right.escapedName);
-                const rightType = getTypeOfSymbol(right);
-                if (left && !maybeTypeOfKind(rightType, TypeFlags.Nullable) && !(maybeTypeOfKind(rightType, TypeFlags.AnyOrUnknown) && right.flags & SymbolFlags.Optional)) {

```

Comment:
@sandersn I'm not sure why we were checking nullable and any/unknown flags here. Seems to me any non-optional property that duplicates a property to the left should trigger the error.

Reply:
I think there was a corner case where a type that included `undefined` was actually for an optional property, but wasn't marked as such. For the others, it might be that `unknown` and (sort of) `any` include `undefined`.

We can ship this in the beta and see if people complain. It's likely that with better optionality tracking, this is not a problem.